main.cpp

```diff
@@ -166,79 +195,123 @@ struct BurgerShack
         void getExtraNapkins(int numNapkins);
     };
 
-    // serve a burger w/fries
     void serveBurger(Customer customer, bool withFries);
-    // charge the customer
     float chargeCustomer(Customer customer, float discountPercentage);
-    // clean the grill
     void cleanGrill(bool extraMessToClean);
 
     Customer nextCustomerInLine;
 };
 
-/*
-Thing 2) Fishing Trawler
-5 properties:
-    1) number of fishing nets on board
-    2) number of crew members on board
-    3) amount of fish caught per day
-    4) amount of gas remaining
-    5) distance covered per week
-3 things it can do:
-    1) pull in the nets
-    2) turn to port
-    3) clean the barnacles off of the hull
- */
+void BurgerShack::serveBurger(Customer customer, bool withFries)
+{
+    std::cout << "Served burger to " << customer.name << std::endl;
+    if(withFries)
+    {
+        std::cout << "with a side of fries" << std::endl;
+    }
+}
+
+float BurgerShack::chargeCustomer(Customer customer, float discountPercentage)
+{
+    float costOfMeal = 5.75f;
+    float amountToCharge = costOfMeal - (costOfMeal * discountPercentage);
 
+    std::cout << "Charged " << customer.name << amountToCharge << "dollars";
+    return amountToCharge;
+}
+
+void cleanGrill(bool extraMessToClean)
+{
+    std::cout << "Cleaned the grill..." << std::endl;
+
+    if(extraMessToClean)
+    {
+        std::cout << "...it was really gross." << std::endl;   
+    }
+}
+
+
+void BurgerShack::Customer::payForMeal(float amountOwed, bool tip)
+{
+    if(tip)
+    {
+        amountOwed += amountOwed * 0.15f;
+    }
+
+    std::cout << "Paid " << amountOwed << "dollars" << std::endl;
+}
+
+float BurgerShack::Customer::checkHowMuchSodaLeftInCup(bool removeLid)
+{
+    float estimatedAmountOfSodaLeft;
+
+    if(removeLid)
+    {
+        estimatedAmountOfSodaLeft = 0.231f;
+    }
+    else
+    {
+        estimatedAmountOfSodaLeft = 0.25f;
+    }
```

Comment:
use a ternary expression to assign to your variable immediately upon construction: 
```
float variable = ( condition ? a : b );
```
if `condition` is true, then `a` is what is assigned. 
if `condition` is false, then `b` is what is assigned. 

- [ ] revise